src/java/org/apache/cassandra/service/accord/AccordConfigurationService.java

```diff
@@ -547,14 +488,15 @@ protected void localSyncComplete(Topology topology, boolean startSync)
         if (!startSync || epochState.syncStatus != SyncStatus.NOT_STARTED)
             return;
 
-        Set<Node.Id> notify = topology.nodes().stream().filter(i -> !localId.equals(i)).collect(Collectors.toSet());
         synchronized (this)
         {
             if (epochState.syncStatus != SyncStatus.NOT_STARTED)
                 return;
-            diskState = diskStateManager.setNotifyingLocalSync(epoch, notify, diskState);
+            diskStateManager.setNotifyingLocalSync(epoch);
             epochState.setSyncStatus(SyncStatus.NOTIFYING);
         }
+
+        Set<Node.Id> notify = topology.nodes().stream().filter(i -> !localId.equals(i)).collect(Collectors.toSet());
```

Comment:
maybe just create the Set with all entries and remove the one we don't want. If we build on my new patch, we can use SortedArraySet to build this usually for almost free, and can even introduce a SortedArraySet.all(SortedArrayList) method that will build a new one with the entries in topology.nodes() extremely cheaply.